ls1intum/Artemis
# `Development`: Remove unnecessary server communication in course overview

### Checklist
#### General


#### Client

### Motivation and Context


Improve performance by getting rid of unnecessary rest Calls in the course overview.

### Description

This PR replaces the for-dashboard calls for discussions by calls to the storage service in the client. Additionally it makes sure that the call is only executed once in the overview.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Course with enabled discussions
- 1 Exercise
- 1 Lecture

1. Log in to Artemis
2. Participate in a discussion in the exercise and the lecture
3. Make sure that the discussions tab is working fine
4. Check that refreshing the course is working fine
5. Make sure that only one for-dashboard call is executed when loading a course and one additional for each refresh. (The one loading all courses on the start page does not count)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| course-management.service.ts | 87% | ✅                        |
| course-discussion.component.ts   | 93% | ✅                        |
